### PR TITLE
Debug last 2

### PR DESCRIPTION
**Pull request by fentontaylor**
Here's a solution for best_fans. The problem I had was that I didn't read the method description carefully enough. Needed to find the biggest *difference* between home win percentage and away win percentage. I wrote the helper methods `home_wins_by_id`, `away_wins_by_id`,  `home_away_win_pct`. We have a little bit of repetition with previous code, but we can clean that up in this PR or the next.

### CODE REVIEW

File: modules/league_statistics.rb [hunk 1]
```diff
@@ -164,12 +164,46 @@ def away_wins_per_team
...
+      percents[team] = {home_pct: 0, away_pct: 0}
+      percents[team][:home_pct] = home_wins[team][:h_wins] / home_wins[team][:h_games].to_f
+      percents[team][:away_pct] = away_wins[team][:a_wins] / away_wins[team][:a_games].to_f
+    end
+    percents
+  end
+
+  def best_fans
+    best = home_away_win_pct
+      .transform_values { |info| info[:home_pct] - info[:away_pct]}
```
Comment (nancylee713) [inline modules/league_statistics.rb]: This method is quite transformative!

**Comment by nancylee713** [comment]
LGTM! 
Corina, you can merge this branch after you review the code

**Comment by fentontaylor** [comment]
This branch contains both the fixes for winningest_team and best_fans now.

**Comment by StarPerfect** [comment]
Awesome! I love how we all continue to work hard on finding solutions even to our most difficult problems! Great job team!

**Pull request merged**